### PR TITLE
Add Note/Announcement about Rename to Litestar

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

[//]: # "Please review the [Starlite contribution guidelines](https://github.com/starlite-api/starlite/blob/main/CONTRIBUTING.rst) for this repository."

- [x] New code has 100% test coverage
- [x] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [x] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

By submitting this issue, you agree to:

- follow Starlite's [Code of Conduct](https://github.com/starlite-api/.github/blob/main/CODE_OF_CONDUCT.md)
- follow Starlite's [Contribution Guidelines](https://starliteproject.dev/community/contribution-guide)

### Description

[//]: # "Please describe your pull request for new release changelog purposes"
Adds a temporary announcement in the docs and `README.md` that notifies users of the upcoming change.

### Close Issue(s)

[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

Closes https://github.com/starlite-api/starlite/issues/1437

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,9 @@
 
 </div>
 
-# Starlite
+# Starlite → Litestar
+
+## [We are renaming to Litestar](https://twitter.com/StarliteAPI/status/1642958305231265792?s=20)
```
Comment (cofin) [inline README.md:36]: We should probably link to something that gives more context about the decision?

**Review by cofin — APPROVED** [review]
LGTM - I made a note re: additional info additions.  That can happen in this PR or a separate one.

File: README.md (hunk 1 above)
Comment (JacobCoffee) [reply README.md:36]: Maybe this should be [the post on r/python](https://www.reddit.com/r/Python/comments/129n4dc/renaming_starlite_to_litestar/) or an even more detailed post on our own subreddit?

**Review by peterschutt — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (JacobCoffee) [reply README.md]: Added reddit post that has more context

**Pull request merged**